File: backend/crm-business/src/main/java/com/crm/business/entity/SignIn.java
package com.crm.business.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 外勤签到实体类
 *
 * @author CRM
 */
@Data
@TableName("bus_sign_in")
public class SignIn {

    /** 签到ID */
    @TableId(type = IdType.AUTO)
    private Long id;

    /** 签到人ID */
    @TableField("user_id")
    private Long userId;

    /** 关联客户ID（拜访签到时填写） */
    @TableField("customer_id")
    private Long customerId;

    /** 客户名称（冗余） */
    @TableField("customer_name")
    private String customerName;

    /** 签到类型 1上午签到 2下午签退 3拜访签到 */
    @TableField("sign_type")
    private Integer signType;

    /** 纬度 */
    @TableField("latitude")
    private BigDecimal latitude;

    /** 经度 */
    @TableField("longitude")
    private BigDecimal longitude;

    /** 签到地址 */
    @TableField("address")
    private String address;

    /** 备注 */
    @TableField("remark")
    private String remark;

    /** 签到时间 */
    @TableField("sign_time")
    private LocalDateTime signTime;

    /** 创建时间 */
    @TableField("create_time")
    private LocalDateTime createTime;
}

File: frontend/src/api/signIn.ts
import request from './request'

/** 签到类型 */
export type SignType = 1 | 2 | 3  // 1上午签到 2下午签退 3拜访签到

/** 签到记录 */
export interface SignIn {
  id?: number
  userId?: number
  customerId?: number | null
  customerName?: string | null
  signType: SignType
  latitude?: number | null
  longitude?: number | null
  address?: string | null
  remark?: string | null
  signTime?: string
  createTime?: string
}

/** 签到参数 */
export interface SignInParams {
  signType: SignType
  latitude?: number
  longitude?: number
  address?: string
  customerId?: number
  customerName?: string
  remark?: string
}

/** 签到打卡 */
export function signIn(params: SignInParams) {
  return request({ url: '/business/sign-in', method: 'post', data: params })
}

/** 查询今日签到记录 */
export function getTodaySignIn() {
  return request({ url: '/business/sign-in/today', method: 'get' })
}

/** 查询签到记录 */
export function getSignInList(startDate?: string, endDate?: string) {
  return request({ url: '/business/sign-in', method: 'get', params: { startDate, endDate } })
}
